### ISSUE TITLE
Async functions that return booleans: un-awaited function calls are not flagged as always being truthy

### ISSUE THREAD

**Issue by AleksandrHovhannisyan** [microsoft/TypeScript#41716]
**TypeScript Version:**  4.1.2

**Search Terms:** async functions, booleans, async return values, async booleans

**Code**

```ts
const isBroken = async () => false;

if (isBroken()) {
  console.log('asd')
}
```

**Expected behavior:**

TypeScript should warn you that the expression evaluated in the `if` statement will always be `true`. [See my explanation of this behavior here](https://www.aleksandrhovhannisyan.com/blog/dev/async-functions-that-return-booleans/). 

If you remove the function call and just do `if (isBroken)`, TypeScript will rightly complain that the expression will always return `true`:

![image](https://user-images.githubusercontent.com/19352442/100524781-f0c9c380-3188-11eb-805a-803de63d5e9f.png)

That's because functions are truthy objects, so when they're coerced to booleans, they're always true. Likewise, an async function returns a Promise object implicitly, and that's always truthy. So TypeScript should really flag that as an error, but it doesn't.

**Actual behavior:**

TypeScript does not complain. The code can lead to bugs, but it's something that can be checked statically before the code is ever compiled.

**Playground Link:**

https://www.typescriptlang.org/play?#code/MYewdgzgLgBAlhAQgJxAawKZhgXhgQwgE8xgYAKASlwD4YAzfAGwgwG4AoDueihFdFmoBvDjBihIIJhgB0TEAHNyAckIATFZQ4BfIA
**Related Issues:**

**Comment by MartinJohns** [comment]
Duplicate of #25330.

**Comment by AleksandrHovhannisyan** [comment]
@MartinJohns My bad, closed
